### PR TITLE
supervision in AdapterClusterImpl

### PR DESCRIPTION
**Pull request by patriknw**
* methods like cluster.join and cluster.subscribe may throw exception if called
  with invalid parameters
* the manager and subscriptions actors should survive such exceptions

### CODE REVIEW

File: akka-cluster-typed/src/main/scala/akka/cluster/typed/internal/AdaptedClusterImpl.scala [hunk 1]
```diff
@@ -147,9 +150,17 @@ private[akka] final class AdapterClusterImpl(system: ActorSystem[_]) extends Clu
...
+      "clusterStateSubscriptions",
+      Props.empty)
 
   override lazy val manager: ActorRef[ClusterCommand] =
-    system.internalSystemActorOf(managerBehavior(classicCluster), "clusterCommandManager", Props.empty)
+    system.internalSystemActorOf(
+      // restart supervision: no state lost in case of failure
+      Behaviors.supervise(managerBehavior(classicCluster)).onFailure(SupervisorStrategy.restart),
+      "clusterCommandManager",
+      Props.empty)
```
Comment (patriknw) [inline akka-cluster-typed/src/main/scala/akka/cluster/typed/internal/AdaptedClusterImpl.scala]: this is the real change, rest is just indentation

Comment (helena) [reply akka-cluster-typed/src/main/scala/akka/cluster/typed/internal/AdaptedClusterImpl.scala]: Nice addition.

**Review by helena — APPROVED** [review]
LGTM

**Review by chbatey — APPROVED** [review]
LGTM

**Pull request merged**